hibernate-reactive-core/src/test/java/org/hibernate/reactive/NaturalIdTest.java

```diff
@@ -0,0 +1,66 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import io.vertx.ext.unit.TestContext;
+import org.hibernate.annotations.NaturalId;
+import org.hibernate.cfg.Configuration;
+import org.junit.Test;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+import static org.hibernate.reactive.common.Identifier.composite;
+import static org.hibernate.reactive.common.Identifier.id;
+
+public class NaturalIdTest extends BaseReactiveTest {
+
+    @Override
+    protected Configuration constructConfiguration() {
+        Configuration configuration = super.constructConfiguration();
+        configuration.addAnnotatedClass(Thing.class);
+        return configuration;
+    }
+
+    @Test
+    public void test(TestContext context) {
+        Thing thing1 = new Thing();
+        thing1.naturalKey = "abc123";
+        thing1.version = 1;
+        Thing thing2 = new Thing();
+        thing2.naturalKey = "abc123";
+        thing2.version = 2;
+        test( context,
+                getSessionFactory()
+                        .withSession( session -> session.persist(thing1, thing2).thenCompose( v -> session.flush() ) )
+                        .thenCompose( v -> getSessionFactory().withSession(
+                                session -> session.find( Thing.class, composite(
+                                        id("naturalKey", "abc123"),
+                                        id("version", 1)
+                                ) )
```

Comment:
Alright, I've now got an API for this which isn't *terrible*, though it's also far from perfect. (It's better if you use the typesafe JPA metamodel.)

I propose including this in the release, and I've clearly marked it `@Incubating`.

Reply:
@DavideD? @Sanne?

Reply:
LGTM

Reply:
I guess an alternative would be a DSL:
```
session.find( Thing.class, id("naturalKey", "abc123")
                                 .and("version", 1) )
```

but I don't think I like this solution more

Reply:
I don't much like it either, even though it's perhaps a little more ergonomic.